backend/routes/admin_routes.py

```
@@ -186,21 +201,37 @@ def create_admin_user():
 #pylint: disable=W0703,R0911
     except IntegrityError:
         db_session.rollback()
-        logging.exception("Database integrity error during user creation")
+        logger.error("User creation failed due to integrity constraint", extra=safe_extra_fields({
+            'username': data.get('username'),
+            'email': data.get('email'),
+            'error_type': 'integrity_constraint',
+            'error_details': str(e),
```

Ruff (0.13.1)

Undefined name `e`

(F821)